Build failure: questionary

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

1. use latest nixos-unstable
2. try and build questionary
3. fails pytest phase

### Can Hydra reproduce this build failure?

Hydra is currently rebuilding this package.

### Link to Hydra build job

https://hydra.nixos.org/build/309465304

### Relevant log output

```console
error: builder for '/nix/store/k081pi7izaml46df7br0hs3k09qfp4di-python3.13-questionary-2.1.1.drv' failed with exit code 1;
       last 25 log lines:
       >                 bold=True,
       >                 underline=False,
       >                 italic=True,
       >                 blink=False,
       >                 reverse=False,
       >                 hidden=False,
       >             )
       >         else:
       > >           assert mock.method_calls[0][1][0] == Attrs(
       >                 color="8b0000",
       >                 bgcolor="",
       >                 bold=True,
       >                 underline=False,
       >                 italic=True,
       >                 blink=False,
       >                 reverse=False,
       >                 hidden=False,
       >                 strike=False,
       >             )
       > E           TypeError: Attrs.__new__() missing 1 required positional argument: 'dim'
       >
       > tests/prompts/test_common.py:220: TypeError
       > =========================== short test summary info ============================
       > FAILED tests/prompts/test_common.py::test_print_with_style - TypeError: Attrs.__new__() missing 1 required positional argument: 'dim'
       > ================= 1 failed, 158 passed, 1 deselected in 13.35s =================
```

### Additional context

_No response_

### System metadata

- system: `"x86_64-linux"`
 - host os: `Linux 6.16.9, NixOS, 25.11 (Xantusia), 25.11pre-git`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.94.0-dev
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/cdockter/.config/nix/nix.conf:/nix/store/lpn2kd6x6f3qnljnrm2c94y6ykys2bph-stylix-kde-config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/cdockter/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/cdockter/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/cdockter/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/9a83ggmdgdbabzyv20fjm130clq873vg-lix-2.94.0-pre-20250912_d90e4a65812c/share`
 - nixpkgs: `not found`

### Notify maintainers

@fabaff

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/co/commitizen/package.nix
+++ b/pkgs/by-name/co/commitizen/package.nix
@@ -1,15 +1,31 @@
 {
   lib,
+  python3,
+  fetchPypi,
   fetchFromGitHub,
   gitMinimal,
   stdenv,
   installShellFiles,
   nix-update-script,
-  python3Packages,
   versionCheckHook,
   writableTmpDirAsHomeHook,
 }:
-
+let
+  # commitizen 4.9.1 is not compatible with version 3.0.52 of prompt-toolkit
+  python = python3.override {
+    packageOverrides = self: super: {
+      prompt-toolkit = super.prompt-toolkit.overridePythonAttrs (oldAttrs: rec {
+        version = "3.0.51";
+        pname = "prompt_toolkit";
+        src = fetchPypi {
+          inherit pname version;
+          hash = "sha256-kxoWLjsn/JDIbxtIux+yxSjCdhR15XycBt4TMRx7VO0=";
+        };
+      });
+    };
+  };
+  python3Packages = python.pkgs;
+in
 python3Packages.buildPythonPackage rec {
   pname = "commitizen";
   version = "4.9.1";
@@ -41,6 +57,7 @@ python3Packages.buildPythonPackage rec {
     importlib-metadata
     jinja2
     packaging
+    prompt-toolkit
     pyyaml
     questionary
     termcolor

--- a/pkgs/development/python-modules/kantoku/default.nix
+++ b/pkgs/development/python-modules/kantoku/default.nix
@@ -1,5 +1,6 @@
 {
   lib,
+  stdenv,
   buildPythonPackage,
   fetchFromGitHub,
   flit-core,
@@ -46,7 +47,14 @@ buildPythonPackage rec {
   disabledTests = [
     # AssertionError
     "test_streams"
+  ]
+  ++ lib.optionals stdenv.hostPlatform.isDarwin [
+    # Assertion error when test_socketstats hits a permission error
+    "test_resource_watcher_max_mem"
+    "test_resource_watcher_max_mem_abs"
   ];
+
+  __darwinAllowLocalNetworking = true;
 
   meta = {
     description = "A Process & Socket Manager built with zmq";

--- a/pkgs/development/python-modules/questionary/default.nix
+++ b/pkgs/development/python-modules/questionary/default.nix
@@ -37,6 +37,7 @@ buildPythonPackage rec {
     "test_blank_line_fix"
 
     # TypeError: Attrs.__new__() missing 1 required positional argument: 'dim'
+    # https://github.com/tmbo/questionary/issues/461
     "test_print_with_style"
   ];
 

